contrib/thrift-cmake/CMakeLists.txt

```diff
@@ -91,5 +92,5 @@ include_directories("${CMAKE_CURRENT_BINARY_DIR}")
 
 add_library(_thrift ${thriftcpp_SOURCES} ${thriftcpp_threads_SOURCES})
 add_library(ch_contrib::thrift ALIAS _thrift)
-target_include_directories(_thrift SYSTEM PUBLIC "${ClickHouse_SOURCE_DIR}/contrib/thrift/lib/cpp/src" ${CMAKE_CURRENT_BINARY_DIR})
+target_include_directories(_thrift SYSTEM PUBLIC "${ClickHouse_SOURCE_DIR}/contrib/thrift/lib/cpp/src" ${CMAKE_CURRENT_BINARY_DIR} "${ClickHouse_SOURCE_DIR}/contrib/boringssl/include")
```

Comment:
The goal is to make the headers of boringssl visible to thrift, right? It is currently done in a rather direct way by referencing the header directory directly. If this directory shall ever move, thrift's build will break.

As a less fragile and more CMake-like alternative, we could exploit the fact that boringssl makes its header files transitively available to libraries which link it (`PUBLIC`):

```
# contrib/boringssl-cmake/CMakeLists.txt:

...
target_include_directories(_crypto SYSTEM PUBLIC "${BORINGSSL_SOURCE_DIR}/include")
target_include_directories(_ssl SYSTEM PUBLIC "${BORINGSSL_SOURCE_DIR}/include")
...
```

So we could basically do the following here:

```
target_include_directories(_thrift SYSTEM PUBLIC "${ClickHouse_SOURCE_DIR}/contrib/thrift/lib/cpp/src" ${CMAKE_CURRENT_BINARY_DIR})   # --> don't add the path manually
target_link_libraries (_thrift PUBLIC OpenSSL::SSL)    # link thrift to boringssl
```